### PR TITLE
Add name field for editing environments after creation

### PR DESCRIPTION
**Pull request by SaiDadireddy**
Related to the Connection Node Failover backend PR -> https://github.com/bastionzero/webshell-backend/pull/1636
Adds `Name` to the modify environment request struct, which allows changing the name after creation as long as the new name is still unique.

### CODE REVIEW

File: bastionzero/service/environments/environments.go
Comment (ymarcus93) [inline bastionzero/service/environments/environments.go:41]: This API change is technically related to the Terraform provider. Once we merge this PR, we can also update the Terraform provider to permit the `bastionzero_environment` resource's name to change without requiring a replace (current behavior).

We can either add this enhancement to the TF provider now, or make a JIRA ticket and add it later. It's a pretty small change to make (with only 1 required integration test to test it), but I'm fine with coming back to it later if there are other higher priority things.

The changes would look something like this:

1) Remove the `RequiresReplace` plan modifier to permit name to be modified without having to destroy and recreate the environment: https://github.com/bastionzero/terraform-provider-bastionzero/blob/ef3edad008c362b536eb370f4130a2ec4675224c/bastionzero/environment/environment.go#L118-L120
2) Add support for modifying `Name` in the `Update` method: https://github.com/bastionzero/terraform-provider-bastionzero/blob/ef3edad008c362b536eb370f4130a2ec4675224c/bastionzero/environment/resource_environment.go#L155-L163
3) Create a new integration test similar to this one here to test updating an environment's name via the TF provider: https://github.com/bastionzero/terraform-provider-bastionzero/blob/ef3edad008c362b536eb370f4130a2ec4675224c/bastionzero/policy/sessionrecording/resource_sessionrecordingpolicy_test.go#L93-L136

File: .changes/unreleased/ENHANCEMENTS-20230725-142941.yaml
Comment (SaiDadireddy) [reply .changes/unreleased/ENHANCEMENTS-20230725-142941.yaml]: [ee4c3bd](https://github.com/bastionzero/bastionzero-sdk-go/pull/26/commits/ee4c3bd1e1e2e910076dfeeece6272e7bac5ec26)

File: .changes/unreleased/ENHANCEMENTS-20230725-142941.yaml
Comment (SaiDadireddy) [reply .changes/unreleased/ENHANCEMENTS-20230725-142941.yaml]: [ee4c3bd](https://github.com/bastionzero/bastionzero-sdk-go/pull/26/commits/ee4c3bd1e1e2e910076dfeeece6272e7bac5ec26)

File: bastionzero/service/environments/environments.go
Comment (SaiDadireddy) [reply bastionzero/service/environments/environments.go:41]: Pull request opened here -> https://github.com/bastionzero/terraform-provider-bastionzero/pull/35

**Review by ymarcus93 — APPROVED** [review]
LGTM. Will merge and release once this feature hits backend in prod. We can also do it when it hits staging and coordinate a Terraform provider release that updates the dependency to new version of the SDK.

**Pull request merged**
